osv/testdata/packagist_test_cases.txt

```diff
@@ -0,0 +1,61 @@
+1 < 2
+10 > 2
+1.0 < 1.1
+1.2 > 1.0.1
+1.0-dev = 1.0-dev
+1.0-dev < 1.0a1
+1.0-dev < 1.0b1
+1.0-dev < 1.0RC1
+1.0-dev < 1.0rc1
+1.0-dev < 1.0
+1.0-dev < 1.0pl1
+1.0a1 > 1.0-dev
+1.0a1 = 1.0a1
+1.0a1 < 1.0b1
+1.0a1 < 1.0RC1
+1.0a1 < 1.0rc1
+1.0a1 < 1.0
+1.0a1 < 1.0pl1
+1.0b1 > 1.0-dev
+1.0b1 > 1.0a1
+1.0b1 = 1.0b1
+1.0b1 < 1.0RC1
+1.0b1 < 1.0rc1
+1.0b1 < 1.0
+1.0b1 < 1.0pl1
+1.0RC1 > 1.0-dev
+1.0RC1 > 1.0a1
+1.0RC1 > 1.0b1
+1.0RC1 = 1.0RC1
+1.0RC1 = 1.0rc1
+1.0RC1 < 1.0
+1.0RC1 < 1.0pl1
+1.0rc1 > 1.0-dev
+1.0rc1 > 1.0a1
+1.0rc1 > 1.0b1
+1.0rc1 = 1.0RC1
+1.0rc1 = 1.0rc1
+1.0rc1 < 1.0
+1.0rc1 < 1.0pl1
+1.0 > 1.0-dev
+1.0 > 1.0a1
+1.0 > 1.0b1
+1.0 > 1.0RC1
+1.0 > 1.0rc1
+1.0 = 1.0
+1.0 < 1.0pl1
+1.0pl1 > 1.0-dev
+1.0pl1 > 1.0a1
+1.0pl1 > 1.0b1
+1.0pl1 > 1.0RC1
+1.0pl1 > 1.0rc1
+1.0pl1 > 1.0
+1.0pl1 = 1.0pl1
+1.25.0 < 1.26.0
+1.0.0 < 1.2-dev
+dev-foo < 1.26.0
```

Comment:
How about dev-foo, dev-master etc? 

https://github.com/composer/semver/blob/main/tests/ComparatorTest.php#L166. Can we just copy every single one from here rather then picking select ones? 

Can we also add a note somewhere (can be just the python code that references this file) about where we pulled these testcases from?

Reply:
Most of them are duplicates because they are also testing the different `> < = <>`...etc. Which means they also need to test when they are `false` instead of `true`. 

The dev-foo to dev-master seems to be a weird case, according to the tests it is neither less than, greater than, nor equal. Not sure what the desired behavior is here.

Reply:
We should match that behaviour, if it's easy to do so. 

For enumeration though, we should filter these out.

Reply:
Not 100% sure how to match that behavior and have it make sense in how we are using the `sort_version` function. For enumeration I added a TODO as I'm not sure exactly what these versions will look like for actual packages, so let's just monitor it for now and implement filtering when we find some on actual packages.